src/lang.rs

```diff
@@ -1101,17 +1162,70 @@ fn var_list(input: Span) -> IResult<Span, Vec<String>> {
 fn parse(input: Span) -> IResult<Span, Positioned<InlineOperator>> {
     with_pos(
         tuple((
-            tag("parse").precedes(multispace1.precedes(req_quoted_string)),
+            tag("parse").precedes(multispace1),
+            opt(tag("regex").precedes(multispace1)),
+            with_pos(req_quoted_string),
             opt(multispace1.precedes(with_pos(pair(tag("from"), multispace1).precedes(expr)))),
-            tag("as").preceded_by(multispace1).precedes(var_list),
+            opt(with_pos(tag("as").preceded_by(multispace1).precedes(var_list))),
             opt(multispace1.precedes(with_pos(pair(tag("from"), multispace1).precedes(expr)))),
             opt(tag("nodrop").preceded_by(multispace1)).map(|nd| nd.is_some()),
         ))
-        .map(|(s, e1, fields, e2, no_drop)| InlineOperator::Parse {
-            pattern: Keyword::new_wildcard(s),
-            fields,
-            input_column: (e1, e2),
-            no_drop,
+        .map(|(_p, is_regex, s, e1, user_fields_opt, e2, no_drop)| {
```

Comment:
can you rename `e2` to `from_col` or something like that? this was a little hard to follow